contracts/modules/STO/Capped/CappedSTO.sol

```diff
@@ -258,4 +273,26 @@ contract CappedSTO is CappedSTOStorage, STO, ReentrancyGuard {
         polyToken.transferFrom(_beneficiary, _to, _fundsAmount);
     }
 
+    /**
+     * @notice Finalizes the STO and mint remaining tokens to treasury address
+     * @notice Treasury wallet address must be whitelisted to successfully finalize
+     */
+    function finalize() external withPerm(ADMIN){
+        require(!isFinalized, "STO is finalized");
+        isFinalized = true;
```

Comment:
Should we add `require(!isFinalized)` to `_preValidatePurchase`? I don't think you can buy tokens once the STO has been finalized.

Reply:
Fixed